### PR TITLE
add documentation on github ssh authentication

### PR DESCRIPTION
**Pull request by fosrias**
@brikis98 This PR addresses https://github.com/gruntwork-io/terragrunt/issues/166#issuecomment-301110407 adding documentation on using ssh with terragrunt.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -301,6 +301,25 @@ In particular:
 
 
 
+#### Github using SSH
+To use terragrunt with SSH github authentication, use the following syntax. Note the `/` after 
+github.com instead of `:` and the uses of `//`.
```
Comment (brikis98) [inline README.md]: I would change the heading title to "Using Terragrunt with private Git repos" and the first sentence to say: 

```
The easiest way to use Terragrunt with private Git repos is to use SSH authentication. Configure your Git account so you can use it with SSH (see the [guide for GitHub here](https://help.github.com/articles/connecting-to-github-with-ssh/) and use the SSH URL for your repo, prepended with `git::ssh://`:
```

File: README.md [hunk 2]
```diff
@@ -301,6 +301,25 @@ In particular:
...
+github.com instead of `:` and the uses of `//`. 
+
+```hcl
+terragrunt = {
+  terraform {
+    source = "git::ssh://git@github.com/foo/modules.git//path/to/module?ref=v0.0.1"
+  }
+}
+```
+For private git repositories, replace github.com with the private repository host.
```
Comment (brikis98) [inline README.md]: I would replace this sentence with "Look up the Git repo for your repository to find the proper format." After all, GitHub has private repos too, and other types of private Git repos may have significantly different URL formats. For example, for private BitBucket repos, you may need to use a URL like `git::ssh://git@my-bitbucket-account.com:7999/team-name/repo-name.git//path/to/module?ref=v0.0.1`.

Comment (fosrias) [inline README.md]: @brikis98 I think I should leave in the comment about `:` as the point of this extra documentation is  partially that using it does not work after the host and your bitbucket example here included it ;). Past that will update later today.

Comment (brikis98) [inline README.md]: Which `:`? The one before the port number? If so, that definitely works with BitBucket...

Comment (fosrias) [inline README.md]: K. I will play with it a bit and double check. IIRC, it did not with github.

Comment (fosrias) [inline README.md]: Ah. Nm. Missed that it was a port. That should work. Problem is `:` vs `/` for path in normal github ssh path.

**Comment by fosrias** [comment]
@brikis98 Comments addressed.

**Comment by brikis98** [comment]
Wonderful, thanks!

**Pull request merged**